public/app/features/alerting/unified/components/silences/SilencesFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, FormEvent } from 'react';
+import { css } from '@emotion/css';
+import { Label, Icon, Input, Tooltip, RadioButtonGroup, useStyles2, Button } from '@grafana/ui';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import { getFiltersFromUrlParams } from '../../utils/misc';
+import { debounce } from 'lodash';
+import { SilenceState } from 'app/plugins/datasource/alertmanager/types';
+
+export const SilencesFilter = () => {
+  const [filterKey, setFilterKey] = useState<number>(Math.floor(Math.random() * 100));
+  const queryStringKey = `queryStringKey-${filterKey}`;
+  const [queryParams, setQueryParams] = useQueryParams();
+  const { queryString, silenceState } = getFiltersFromUrlParams(queryParams);
+  const styles = useStyles2(getStyles);
+
+  const handleQueryStringChange = debounce((e: FormEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLInputElement;
+    setQueryParams({ queryString: target.value || null });
+  }, 600);
+
+  const handleSilenceStateChange = (state: string) => {
+    setQueryParams({ silenceState: state });
+  };
+
+  const clearFilters = () => {
+    setQueryParams({
+      queryString: null,
+      silenceState: null,
+    });
+    setTimeout(() => setFilterKey(filterKey + 1), 100);
+  };
+
+  const stateOptions: SelectableValue[] = Object.entries(SilenceState).map(([key, value]) => ({
```

Comment:
`stateOptions` is static as `SilenceState` can never change. Better initialize it outside a component